### PR TITLE
feat: crate query for adding TaskId To All Subscribed Users

### PR DESCRIPTION
**Pull request by Ahmedelsa3eed**
i did not test this query yet, but i need someone to verify that registeredCourse model is working properly

### CODE REVIEW

**Review by Mkhaled707 — COMMENTED** [review]
There is no problem regarding the changes on the registered courses code, instead of one function in the "RegistredCoursesRespository" responsible for deleting a registered course, which I will fix later, you can merge the branch after getting approval from the person responsible for reviewing the study task part.

File: Student-Unity/src/main/java/csed/swe/studentunity/model/RegisteredCourseId.java
Comment (AhmedElemary57) [inline Student-Unity/src/main/java/csed/swe/studentunity/model/RegisteredCourseId.java:15]: Why you have changed the userId to user ???
I now that it is user type but we are only interested in its id and we do so in the contractor

File: Student-Unity/src/main/java/csed/swe/studentunity/model/StudentTaskId.java
Comment (AhmedElemary57) [inline Student-Unity/src/main/java/csed/swe/studentunity/model/StudentTaskId.java:20]: the same here we did changed the studentId to user

File: Student-Unity/src/main/java/csed/swe/studentunity/model/RegisteredCourseId.java
Comment (Mkhaled707) [reply Student-Unity/src/main/java/csed/swe/studentunity/model/RegisteredCourseId.java:15]: I thought so at the beginning so I used the id only like you did in the task, but the problem is if the user got deleted from the users table his registered courses should be deleted too, so we need a foreign key, is the right way of making a foreign key using the long id only or referencing the user object through mapping and "refrencedColumnName"?

Comment (Ahmedelsa3eed) [reply Student-Unity/src/main/java/csed/swe/studentunity/model/RegisteredCourseId.java:15]: to be able to reference the user PK from ``RegisteredCourses`` Entitiy

**Review by AhmedElemary57 — APPROVED** [review]

**Pull request merged**
